app/routers/guest/applications.py

```diff
@@ -230,14 +230,13 @@ async def credit_product_options(
     :raise: HTTPException with status code 404 if the application is expired, not in the ACCEPTED status, or if the
             previous lenders are not found.
     """
-    with transaction_session(session):
+    with rollback_on_error(session):
```

Comment:
I think this `with` can be deleted, since we have no need for a transaction?

Reply:
Yeap, I thought the same, will delete